scripts/globals/pets/avatar.lua

```diff
@@ -0,0 +1,385 @@
+    local master = pet:getMaster()
+    local posTarget = nil
+    local posSpellID = 0
+    local buffTarget = nil
+    local buffSpellID = 0
+    local cureTarget = nil
+    local spellID = 0
+
+    if master then
+        if not pet:isEngaged() then
+            pet:lookAt(master:getPos())
+        else
+            -- don't do logic to buff anyone if pet is engaged. buffTarget will be reset below
+            buffTarget = master
+        end
+
+        local distance  = pet:checkDistance(master) -- starts as distance to master, updated to be distance to the posTarget
+        local hpp       = 0
+        local alliance  = master:getAlliance()
+        local party     = master:getParty()
+        local petLvl    = pet:getMainLvl()
+
+        for _, member in pairs(alliance) do
+            -- tiered chance of healing based on how low they are
+            local tempHPP = member:getHPP()
+            if
+                pet:checkDistance(member) < 20 and
+                (hpp == 0 or
+                tempHPP < hpp) and
+                ((tempHPP < 75 and
+                math.random(100) < 50) or
+                (tempHPP < 50 and
+                math.random(100) < 50) or
+                (tempHPP < 25 and
+                math.random(100) < 50))
+            then
+                hpp = tempHPP
+                cureTarget = member
+            end
+        end
+
+        if cureTarget then
+            -- curaga can be used on anyone in the alliance and isn't determined by how many need a heal (assumed to be based on skill over cap)
+            if
+                petLvl >= 16 and
+                math.random(100) <= 2 * xi.summon.getSummoningSkillOverCap(pet)
+            then
+                -- curaga
+                if petLvl >= 71 then
+                    spellID = xi.magic.spell.CURAGA_IV
+                elseif petLvl >= 51 then
+                    spellID = xi.magic.spell.CURAGA_III
+                elseif petLvl >= 31 then
+                    spellID = xi.magic.spell.CURAGA_II
+                else
+                    spellID = xi.magic.spell.CURAGA
+                end
+            else
+                if petLvl >= 61 then
+                    spellID = xi.magic.spell.CURE_V
+                elseif petLvl >= 41 then
+                    spellID = xi.magic.spell.CURE_IV
+                elseif petLvl >= 21 then
+                    spellID = xi.magic.spell.CURE_III
+                elseif petLvl >= 11 then
+                    spellID = xi.magic.spell.CURE_II
+                else
+                    spellID = xi.magic.spell.CURE
+                end
+            end
+
+            return spellID, cureTarget
+        else
+            local hasteTier = { 0, 0 }
+            if petLvl >= 40 then
+                hasteTier = { 1, 15 }
+            end
+
+            local regenTier = { 0, 0 }
+            if petLvl >= 21 then
+                regenTier = { 1, 1 }
+            end
+
+            local shellTier = { 0, 0 }
```

Comment:
This block looks suspiciously like it can be written as a table+driver

Reply:
Untested. Something like this:
```lua
local petLvl = 10

local tbl =
{
    [68] = { 4, 2617 },
    [52] = { 3, 2617 },
    [37] = { 2, 2617 },
    [17] = { 1, 2617 },
}

-- TODO: Iterate in order so you match highest or lowest, or whatever
for key, val in pairs(tbl) do
    if petLvl >= key then
        shellTier = val
        break
    end
end

```

Reply:
Following up as well on this one, please if you do use the above as an example, don't use key, val as variable names

Reply:
Done. Hopefully you like it. I'm pretty happy of the results. 

I stored the spells in an ordered list (highest level first) using the effect as a key. Loop and break out when a spell is available to the pet.